File: schemaorg-rs/src/statistical_variable.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///https://schema.org/additionalType
///https://schema.org/Text
///https://schema.org/URL
pub type StatisticalVariableAdditionalTypeFieldEnum = String;
///https://schema.org/StatisticalVariable
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct StatisticalVariable {
    #[serde(rename = "@context")]
    pub context: String,
    ///https://schema.org/measurementMethod
    #[serde(rename = "measurementMethod")]
    #[serde_as(as = "OneOrMany<_>")]
    pub measurement_method: Vec<StatisticalVariableMeasurementMethodFieldEnum>,
    ///https://schema.org/measuredProperty
    #[serde(rename = "measuredProperty")]
    #[serde_as(as = "OneOrMany<_>")]
    pub measured_property: Vec<Property>,
    ///https://schema.org/image
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<StatisticalVariableImageFieldEnum>,
    ///https://schema.org/constraintProperty
    #[serde(rename = "constraintProperty")]
    #[serde_as(as = "OneOrMany<_>")]
    pub constraint_property: Vec<StatisticalVariableConstraintPropertyFieldEnum>,
    ///https://schema.org/potentialAction
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///https://schema.org/url
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///https://schema.org/subjectOf
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<StatisticalVariableSubjectOfFieldEnum>,
    ///https://schema.org/numConstraints
    #[serde(rename = "numConstraints")]
    #[serde_as(as = "OneOrMany<_>")]
    pub num_constraints: Vec<i32>,
    ///https://schema.org/mainEntityOfPage
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<StatisticalVariableMainEntityOfPageFieldEnum>,
    ///https://schema.org/measurementDenominator
    #[serde(rename = "measurementDenominator")]
    #[serde_as(as = "OneOrMany<_>")]
    pub measurement_denominator: Vec<StatisticalVariable>,
    ///https://schema.org/measurementTechnique
    #[serde(rename = "measurementTechnique")]
    #[serde_as(as = "OneOrMany<_>")]
    pub measurement_technique: Vec<StatisticalVariableMeasurementTechniqueFieldEnum>,
    ///https://schema.org/additionalType
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<StatisticalVariableAdditionalTypeFieldEnum>,
    ///https://schema.org/sameAs
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///https://schema.org/populationType
    #[serde(rename = "populationType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub population_type: Vec<Class>,
    ///https://schema.org/statType
    #[serde(rename = "statType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub stat_type: Vec<StatisticalVariableStatTypeFieldEnum>,
    ///https://schema.org/alternateName
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///https://schema.org/description
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<StatisticalVariableDescriptionFieldEnum>,
    ///https://schema.org/measurementQualifier
    #[serde(rename = "measurementQualifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub measurement_qualifier: Vec<Enumeration>,
    ///https://schema.org/disambiguatingDescription
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///https://schema.org/identifier
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<StatisticalVariableIdentifierFieldEnum>,
    ///https://schema.org/name
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
}

File: schemaorg-ts/src/statistical_variable.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type StatisticalVariableAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/StatisticalVariable>
 **/
export interface StatisticalVariable {
  "@context": string;
  "@type": "StatisticalVariable";

  /**
   * <https://schema.org/measurementDenominator>
   **/
  measurementDenominator: all.OneOrMany<all.StatisticalVariable>;

  /**
   * <https://schema.org/populationType>
   **/
  populationType: all.OneOrMany<all.Class>;

  /**
   * <https://schema.org/measurementQualifier>
   **/
  measurementQualifier: all.OneOrMany<all.Enumeration>;

  /**
   * <https://schema.org/statType>
   **/
  statType: all.OneOrMany<all.StatisticalVariableStatTypeFieldEnum>;

  /**
   * <https://schema.org/measuredProperty>
   **/
  measuredProperty: all.OneOrMany<all.Property>;

  /**
   * <https://schema.org/measurementTechnique>
   **/
  measurementTechnique: all.OneOrMany<all.StatisticalVariableMeasurementTechniqueFieldEnum>;

  /**
   * <https://schema.org/measurementMethod>
   **/
  measurementMethod: all.OneOrMany<all.StatisticalVariableMeasurementMethodFieldEnum>;

  /**
   * <https://schema.org/numConstraints>
   **/
  numConstraints: all.OneOrMany<number>;

  /**
   * <https://schema.org/constraintProperty>
   **/
  constraintProperty: all.OneOrMany<all.StatisticalVariableConstraintPropertyFieldEnum>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.StatisticalVariableMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.StatisticalVariableIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.StatisticalVariableDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.StatisticalVariableSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.StatisticalVariableAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.StatisticalVariableImageFieldEnum>;
}
